modules/apps/style-book/style-book-web/src/main/java/com/liferay/style/book/web/internal/portlet/action/PreviewFragmentCollectionMVCResourceCommand.java

```diff
@@ -90,17 +97,41 @@ private String _renderPortalNormal(
 			(ThemeDisplay)httpServletRequest.getAttribute(
 				WebKeys.THEME_DISPLAY);
 
-		LayoutSet layoutSet = _layoutSetLocalService.getLayoutSet(
-			themeDisplay.getScopeGroupId(), false);
+		Theme theme;
 
-		themeDisplay.setLayoutSet(layoutSet);
-		themeDisplay.setLookAndFeel(
-			layoutSet.getTheme(), layoutSet.getColorScheme());
+		if (!FeatureFlagManagerUtil.isEnabled(
+				themeDisplay.getCompanyId(), "LPD-30204")) {
+
+			LayoutSet layoutSet = _layoutSetLocalService.getLayoutSet(
+				themeDisplay.getScopeGroupId(), false);
+
+			theme = layoutSet.getTheme();
+
+			themeDisplay.setLayoutSet(layoutSet);
+
+			themeDisplay.setLookAndFeel(theme, layoutSet.getColorScheme());
+		}
+		else {
+			long styleBookEntryId = GetterUtil.getLong(
+				httpServletRequest.getParameter("styleBookEntryId"));
+
+			StyleBookEntry styleBookEntry =
+				_styleBookEntryLocalService.fetchStyleBookEntry(
+					styleBookEntryId);
+
+			theme = _themeLocalService.fetchTheme(
```

Comment:
This looks like it won't account for the situation where the style book is based on a themeCSS CX. Is that correct?